driver-core/src/main/com/mongodb/connection/Stream.java

```diff
@@ -60,6 +60,42 @@
      */
     ByteBuf read(int numBytes) throws IOException;
 
+    /**
```

Comment:
This was necessary because Stream is part of the public API, and on the off chance that someone is using a custom implementation of Stream, this will prevent us from using the streaming protocol until that implementation adds support for adding a per-read additional timeout.  All of the built-in implementations will return true for this.